vertical-pod-autoscaler/pkg/target/fetcher.go

```diff
@@ -187,6 +190,12 @@ func getLabelSelector(informer cache.SharedIndexInformer, kind, namespace, name
 			return nil, fmt.Errorf("Failed to parse %s %s/%s", kind, namespace, name)
 		}
 		return metav1.LabelSelectorAsSelector(metav1.SetAsLabelSelector(apiObj.Spec.Selector))
+	case (*v1batch.CronJob):
+		apiObj, ok := obj.(*v1batch.CronJob)
+		if !ok {
+			return nil, fmt.Errorf("Failed to parse %s %s/%s", kind, namespace, name)
+		}
+		return metav1.LabelSelectorAsSelector(metav1.SetAsLabelSelector(apiObj.Spec.JobTemplate.Spec.Selector))
```

Comment:
This will not work in all (most?) cases. Here is a sample CronJob i created with kubectl run command. It doesn't have Selector. Selector for (Cron)Jobs is optional.
```
$  kubectl run --schedule="* * * * *" bb --image=busybox --restart=Never --command -- date
$ kubectl get cronjob -o yaml
apiVersion: v1
items:
- apiVersion: batch/v1beta1
  kind: CronJob
  metadata:
    creationTimestamp: 2019-03-15T12:59:13Z
    labels:
      run: bb
    name: bb
    namespace: default
    resourceVersion: "1162"
    selfLink: /apis/batch/v1beta1/namespaces/default/cronjobs/bb
    uid: 22104d0a-4722-11e9-a206-0242ac110028
  spec:
    concurrencyPolicy: Allow
    failedJobsHistoryLimit: 1
    jobTemplate:
      metadata:
        creationTimestamp: null
      spec:
        template:
          metadata:
            creationTimestamp: null
            labels:
              run: bb
          spec:
            containers:
            - command:
              - date
              image: busybox
              imagePullPolicy: Always
              name: bb
              resources: {}
              terminationMessagePath: /dev/termination-log
              terminationMessagePolicy: File
            dnsPolicy: ClusterFirst
            restartPolicy: Never
            schedulerName: default-scheduler
            securityContext: {}
            terminationGracePeriodSeconds: 30
    schedule: '* * * * *'
    successfulJobsHistoryLimit: 3
    suspend: false
  status:
    lastScheduleTime: 2019-03-15T13:06:00Z
[...]
```

Reply:
Thanks for the review I'll update it accordingly.